chore: FIT-982: sync SDK and clean up public docstrings

robot-ci-heartex opened the pull request:
<!-- Follow Merge -->
This PR was automatically generated via Follow Merge. [Workflow Run](https://github.com/HumanSignal/label-studio/actions/runs/19307976579)
Please ensure that all linked upstream Pull Requests are merged before proceeding with this one.
- https://github.com/HumanSignal/label-studio-sdk/pull/671
<!-- Follow Merge -->

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19301743405

matt-bernstein commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/19307965825)
> Successfully merged: Already up to date.

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/19307976579

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio-enterprise/actions/runs/19329329964
